providers/src/airflow/providers/edge/worker_api/routes/jobs.py

```diff
@@ -0,0 +1,128 @@
+from airflow.utils import timezone
+from airflow.utils.sqlalchemy import with_row_locks
+from airflow.utils.state import TaskInstanceState
+
+jobs_router = AirflowRouter(tags=["Jobs"], prefix="/jobs")
+
+
+@jobs_router.get(
+    "/fetch/{worker_name}",
+    dependencies=[Depends(jwt_token_authorization_rest)],
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_400_BAD_REQUEST,
+            status.HTTP_403_FORBIDDEN,
+        ]
+    ),
+)
+def fetch(
+    worker_name: str,
+    body: Annotated[
+        WorkerQueuesBody,
+        Body(
+            title="Log data chunks",
+            description="The worker remote has no access to log sink and with this can send log chunks to the central site.",
+        ),
+    ],
+    session: Annotated[Session, Depends(get_session)],
+) -> EdgeJobFetched | None:
+    """Fetch a job to execute on the edge worker."""
+    query = (
+        select(EdgeJobModel)
+        .where(EdgeJobModel.state == TaskInstanceState.QUEUED)
+        .order_by(EdgeJobModel.queued_dttm)
+    )
+    if body.queues:
+        query = query.where(EdgeJobModel.queue.in_(body.queues))
+    query = query.limit(1)
+    query = with_row_locks(query, of=EdgeJobModel, session=session, skip_locked=True)
+    job: EdgeJobModel = session.scalar(query)
+    if not job:
+        return None
+    job.state = TaskInstanceState.RUNNING
+    job.edge_worker = worker_name
+    job.last_update = timezone.utcnow()
+    session.commit()
+    return EdgeJobFetched(
+        dag_id=job.dag_id,
+        task_id=job.task_id,
+        run_id=job.run_id,
+        map_index=job.map_index,
+        try_number=job.try_number,
+        command=literal_eval(job.command),
+        concurrency_slots=job.concurrency_slots,
+    )
+
+
+@jobs_router.patch(
+    "/state/{dag_id}/{task_id}/{run_id}/{try_number}/{map_index}/{state}",
+    dependencies=[Depends(jwt_token_authorization_rest)],
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_400_BAD_REQUEST,
+            status.HTTP_403_FORBIDDEN,
+        ]
+    ),
+)
+def state(
+    dag_id: Annotated[str, WorkerApiDocs.dag_id],
+    task_id: Annotated[str, WorkerApiDocs.task_id],
+    run_id: Annotated[str, WorkerApiDocs.run_id],
+    try_number: Annotated[int, WorkerApiDocs.try_number],
+    map_index: Annotated[int, WorkerApiDocs.map_index],
+    state: Annotated[TaskInstanceState, WorkerApiDocs.state],
+    session: Annotated[Session, Depends(get_session)],
+) -> None:
+    """Update the state of a job running on the edge worker."""
+    query = select(EdgeJobModel).where(
+        EdgeJobModel.dag_id == dag_id,
+        EdgeJobModel.task_id == task_id,
+        EdgeJobModel.run_id == run_id,
+        EdgeJobModel.map_index == map_index,
+        EdgeJobModel.try_number == try_number,
+    )
+    job: EdgeJobModel = session.scalar(query)
+    if job:
+        job.state = state
+        job.last_update = timezone.utcnow()
+        session.commit()
```

Comment:
```py
update(EdgeJobModel).where(
        EdgeJobModel.dag_id == dag_id,
        EdgeJobModel.task_id == task_id,
        EdgeJobModel.run_id == run_id,
        EdgeJobModel.map_index == map_index,
        EdgeJobModel.try_number == try_number,
    )
).values(state=state, last_update=timezone.utcnow())
result = session.execute(query)
log.info("State updated: %s row(s) affected", result.rowcount)
```

Reply:
Ah, that is a good thing to optimize. Should have jumped into my eyes earlier!